### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by thusoy**
This should be fine, but while fixing this I noticed the removing backslashes thing, which I believe wasn't hit by tests and might be superfluous with the refactored parsing, you might want to check that out.

### CODE REVIEW

File: mohawk/base.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def _authorize(self, mac_type, parsed_header, resource,
                                                ts=parsed_header['ts'],
                                                id=resource.credentials['id']))
         else:
-            log.warn('seen_nonce was None; not checking nonce. '
+            log.warning('seen_nonce was None; not checking nonce. '
                      'You may be vulnerable to replay attacks')
```
Comment (kumar303) [inline mohawk/base.py]: Sorry for the nitpick but could you align the quotes with the opening parenthesis like this:
````py
log.warning('seen_nonce was None; not checking nonce. '
            'You may be vulnerable to replay attacks')
````

**Comment by thusoy** [comment]
Sure, fixed. (also heads up: you have edit rights as a maintainer, so you can fix small stuff yourself to avoid some of the back and forth).

**Comment by kumar303** [comment]
Oh whoops, I just noticed the failures in Python 3.4 and 3.5 (because of warnings). Maybe the test suite can look for an environment variable like `MOHAWK_ALLOW_WARNINGS` ? I think we'll need to ignore a lot of stuff for newer Pythons.

**Comment by thusoy** [comment]
Well, `assertEquals` has been [deprecated since 3.2](https://docs.python.org/3.6/library/unittest.html#deprecated-aliases), highlighting stuff like that was part of the point of this patch, my initial problem was that I wanted to highlight deprecation warnings in my own code, but kept getting failures from mohawk. Threw up a patch with `s/assertEquals/assertEqual`, let's see what Travis thinks about that, I don't think there's much else.

**Comment by thusoy** [comment]
Mostly happy, but there seems to be an unrelated failure on 2.7, do you know if that one is a bit flaky or something I should investigate?

**Comment by kumar303** [comment]
ugh, yeah, it may be a flaky test. I restarted the job so we'll see in a minute.

**Comment by thusoy** [comment]
Apparently was, Travis is happy camper now.

**Comment by kumar303** [comment]
Cool. If we ever hit harder to fix warnings (because of maintaining backwards compatibility) then we can use an env var to allow warnings. Maybe we won't need to do that though. 

This change is great, thanks. It should help us future proof the code more easily.

**Pull request merged**